From d22eb1f9e2431199f1b6bea2aefe8473284e966e Mon Sep 17 00:00:00 2001
Message-Id: <d22eb1f9e2431199f1b6bea2aefe8473284e966e.1576867467.git.liu.denton@gmail.com>
In-Reply-To: <cover.1576867467.git.liu.denton@gmail.com>
References: <cover.1576531851.git.liu.denton@gmail.com>
	<cover.1576867467.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Mon, 16 Dec 2019 13:07:43 -0800
Subject: [RESEND PATCH 2/2] contrib/credential/netrc: work outside a repo
To: Git Mailing List <git@vger.kernel.org>
Cc: Luis Marsano <luis.marsano@gmail.com>,
    Ted Zlatanov <tzz@lifelogs.com>,
    Jeff King <peff@peff.net>

Currently, git-credential-netrc does not work outside of a git
repository. It fails with the following error:

	fatal: Not a git repository: . at /usr/share/perl5/Git.pm line 214.

There is no real reason why need to be within a repository, though.
Credential helpers should be able to work just fine outside the
repository as well.

Call the non-self version of config() so that git-credential-netrc no
longer needs to be run within a repository.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 contrib/credential/netrc/git-credential-netrc.perl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/credential/netrc/git-credential-netrc.perl b/contrib/credential/netrc/git-credential-netrc.perl
index ebfc123ec6..bc57cc6588 100755
--- a/contrib/credential/netrc/git-credential-netrc.perl
+++ b/contrib/credential/netrc/git-credential-netrc.perl
@@ -423,7 +423,7 @@ sub load_config {
 	# load settings from git config
 	my $options = shift;
 	# set from command argument, gpg.program option, or default to gpg
-	$options->{'gpg'} //= Git->repository()->config('gpg.program')
+	$options->{'gpg'} //= Git::config('gpg.program')
 	                  // 'gpg';
 	log_verbose("using $options{'gpg'} for GPG operations");
 }
-- 
2.24.1.703.g2f499f1283

